app/models/user.rb

```diff
@@ -18,6 +18,21 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  devise :confirmable, :database_authenticatable,
+         :recoverable, :registerable,
+         :rememberable, :validatable
+  validates :email, uniqueness: true
+
+  def ban!
+    raise 'User::AlreadyBanned' if banned?
```

Comment:
This sounds like a constant reference, but is just a string... intentional?

Reply:
Yeah. I didn't really want to make an error. It looked more official than a string saying 'User has already been banned'.